DataScience/Analytics/prototype/landRegistryAddresses.py

```diff
@@ -0,0 +1,109 @@
+
+
+Requirements
+------------
+
+:requires: pandas (0.19.1)
+:requires: addressLinking (and all the requirements within it)
+
+
+Author
+------
+
+:author: Sami Niemi (sami.niemi@valtech.co.uk)
+
+
+Version
+-------
+
+:version: 0.1
+:date: 30-Nov-2016
+
+
+Results
+-------
+
+With full AB and a reasonable runtime (AddressLinker version 0.1):
+    Total Match Fraction 98.4 per cent
+"""
+from Analytics.linking import addressLinking
+import pandas as pd
+
+
+class LandRegistryLinker(addressLinking.AddressLinker):
+    """
+    Address Linker for Land Registry test data. Inherits the AddressLinker and overwrites the load_data method.
+    """
+
+    def load_data(self):
+        """
+        Read in the Land Registry testing data. Note that the data should be preprocessed with a script
+        found in the DataScience/Analytics/data/ folder.
+
+        The data were originally downloaded from:
+        https://data.gov.uk/dataset/land-registry-monthly-price-paid-data
+        The header was grabbed from:
+        https://www.gov.uk/guidance/about-the-price-paid-data#explanations-of-column-headers-in-the-ppd
+        """
+        self.toLinkAddressData = pd.read_csv(self.settings['inputPath'] + self.settings['inputFilename'],
+                                             low_memory=False, parse_dates=[2, ], infer_datetime_format=True)
+
+        # drop unnecessary information
+        self.toLinkAddressData.drop(['Price', 'TransferDate', 'Type', 'New', 'Duration',
+                                     'PPD', 'District', 'RecordStatus'],
+                                    axis=1, inplace=True)
+
+        # fill NaNs with empty strings so that we can form a single address string
+        self.toLinkAddressData.fillna('', inplace=True)
+        self.toLinkAddressData['ADDRESS'] = self.toLinkAddressData['SAO'] + ' ' + \
+                                            self.toLinkAddressData['PAO'] + ' ' + \
+                                            self.toLinkAddressData['Street'] + ' ' + \
+                                            self.toLinkAddressData['Locality'] + ' ' + \
+                                            self.toLinkAddressData['Town'] + ' ' + \
+                                            self.toLinkAddressData['Postcode']
+
+        # rename postcode to postcode_orig and locality to locality_orig
+        self.toLinkAddressData.rename(columns={'Postcode': 'postcode_orig',
+                                               'Locality': 'locality_orig',
+                                               'TransactionID': 'ID'}, inplace=True)
+
+
+def run_land_registry_linker():
+    """
+    A simple wrapper that allows running Land Registry linker.
+
+    :return: None
+    """
+    settings = dict(inputFilename='pp-monthly-update-Edited.csv',
+                    inputPath='/Users/saminiemi/Projects/ONS/AddressIndex/data/',
+                    outname='LandRegistry')
+
+    linker = LandRegistryLinker(**settings)
```

Comment:
Yep it would be better. After this PR I actually need to expose those options to be changeable so that I can run the same code in different hosts without needing to make changes to the paths, so I will need to make some mods to this logic.